sycl/include/sycl/detail/property_helper.hpp

```diff
@@ -88,6 +88,7 @@ class PropertyWithDataBase {
   PropertyWithDataBase(int ID) : MID(ID) {}
   bool isSame(int ID) const { return ID == MID; }
   virtual ~PropertyWithDataBase() = default;
+  int getKind() const { return MID; }
```

Comment:
Nit: Could be made `noexcept` while we are at it.

Reply:
discussed with Steffen that it is probably not needed, impl is too obvious, no benefit from adding noexcept here.